scala StandaloneWSRequest.stream() buffers entire response in memory

As of version `2.2.0`, calling the scala API `StandaloneWSRequest.stream()` now behaves the same as `.execute()`. Both buffer the entire response in memory because a `StandaloneAhcWSResponse` is returned. Previously, `.stream()` returned a `StreamedResponse`. The change in behavior was introduced by removing the call to `.executeStream()` in  https://github.com/playframework/play-ws/commit/00c8ddb3ad278a774970ed0b10c58f04f8726517#diff-cc289ad1c53eab708b302d4dec006082cf9902504ccd91c198ad20de6319296eL239

### Play WS Version 2.2.0

### API Scala

### Operating System ANY

### JDK ANY

### Expected Behavior

Calling `StandaloneWSRequest.stream()` followed by `Response.bodyAsSource` on the response should give the caller access to a stream of bytes which does not require buffering the entire response in memory.

### Actual Behavior

As of version `2.2.0` `StandaloneWSRequest.stream()` and `StandaloneWSRequest.execute()` are essentially the same.

Memory usage increases as the response is processed until the following exception occurs:
```
Caused by: java.nio.BufferOverflowException
at java.base/java.nio.ByteBuffer.put(ByteBuffer.java:914)
at play.shaded.ahc.org.asynchttpclient.netty.NettyResponse.getResponseBodyAsBytes(NettyResponse.java:170)
at play.api.libs.ws.ahc.StandaloneAhcWSResponse.bodyAsBytes(StandaloneAhcWSResponse.scala:48)
at play.api.libs.ws.ahc.StandaloneAhcWSResponse.bodyAsSource(StandaloneAhcWSResponse.scala:50)
```

On `2.1.11` this exception does not occur.

### Reproducible Test Case

I can't really think of an easy way to verify this by just checking the behavior unless there is a way to limit how much memory can be allocated to buffers during the test. I don't have the necessary expertise for that unfortunately. A test could make an assertion that the type returned by `StandaloneWSRequest.stream()` is `StreamedResponse`, but I'm not sure that is appropriate.

--- a/play-ahc-ws-standalone/src/main/scala/play/api/libs/ws/ahc/StandaloneAhcWSRequest.scala
+++ b/play-ahc-ws-standalone/src/main/scala/play/api/libs/ws/ahc/StandaloneAhcWSRequest.scala
@@ -49,7 +49,7 @@ case class StandaloneAhcWSRequest(
     with AhcUtilities
     with WSCookieConverter {
   override type Self     = StandaloneAhcWSRequest
-  override type Response = StandaloneAhcWSResponse
+  override type Response = StandaloneWSResponse
 
   require(client != null, "A StandaloneAhcWSClient is required, but it is null")
   require(url != null, "A url is required, but it is null")
@@ -223,8 +223,6 @@ case class StandaloneAhcWSRequest(
   override def withMethod(method: String): Self = copy(method = method)
 
   private val executor: StandaloneAhcWSRequest => Future[Response] = {
-    import scala.concurrent.ExecutionContext.Implicits.global
-
     lazy val executor = filterWSRequestExecutor(WSRequestExecutor {
       case ahcReq: StandaloneAhcWSRequest =>
         client.execute(ahcReq.buildRequest())
@@ -234,9 +232,7 @@ case class StandaloneAhcWSRequest(
     })
 
     { (req: StandaloneAhcWSRequest) =>
-      executor(req).collect { case resp: StandaloneAhcWSResponse =>
-        resp
-      }
+      executor(req)
     }
   }
 
@@ -246,7 +242,13 @@ case class StandaloneAhcWSRequest(
     filters.foldRight(next)((filter, executor) => filter.apply(executor))
   }
 
-  override def stream(): Future[Response] = executor(this)
+  override def stream(): Future[Response] = {
+    val executor = filterWSRequestExecutor(WSRequestExecutor { request =>
+      client.executeStream(request.asInstanceOf[StandaloneAhcWSRequest].buildRequest())
+    })
+
+    executor(this)
+  }
 
   /**
    * Returns the HTTP header given by name, using the request builder.  This may be signed,

